kusto/ingest/status.go

```diff
@@ -0,0 +1,301 @@
+package ingest
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-kusto-go/kusto/ingest/internal/properties"
+	"github.com/google/uuid"
+	"github.com/kylelemons/godebug/pretty"
+	storageuid "github.com/satori/go.uuid"
+)
+
+// StatusCode is the ingestion status
+type StatusCode string
+
+const (
+	// Pending status represents a temporary status.
+	// Might change during the course of ingestion based on the
+	// outcome of the data ingestion operation into Kusto.
+	Pending StatusCode = "Pending"
+	// Succeeded status represents a permanent status.
+	// The data has been successfully ingested to Kusto.
+	Succeeded StatusCode = "Succeeded"
+	// Failed Status represents a permanent status.
+	// The data has not been ingested to Kusto.
+	Failed StatusCode = "Failed"
+	// Queued status represents a permanent status.
+	// The data has been queued for ingestion &  status tracking was not requested.
+	// (This does not indicate that the ingestion was successful.)
+	Queued StatusCode = "Queued"
+	// Skipped status represents a permanent status.
+	// No data was supplied for ingestion. The ingest operation was skipped.
+	Skipped StatusCode = "Skipped"
+	// PartiallySucceeded status represents a permanent status.
+	// Part of the data was successfully ingested to Kusto, while other parts failed.
+	PartiallySucceeded StatusCode = "PartiallySucceeded"
+
+	// StatusRetrievalFailed means the client ran into truble reading the status from the service
+	StatusRetrievalFailed StatusCode = "StatusRetrievalFailed"
+	// StatusRetrievalCanceled means the user canceld the status check
+	StatusRetrievalCanceled StatusCode = "StatusRetrievalCanceled"
+)
+
+// IsFinal returns true if the ingestion status is a final status, or false if the status is temporary
+func (i StatusCode) IsFinal() bool {
+	return i != Pending
+}
+
+// IsSuccess returns true if the status code is a final successfull status code
+func (i StatusCode) IsSuccess() bool {
+	switch i {
+	case Succeeded:
+		return true
+
+	case Queued:

```

Comment:
I'm not sure why Queued is considered succeeded, unless Queued is guaranteed to always complete with a success status.  That seems hard to guarantee that the backend wouldn't change that behavior.  

Also, this can be made more succinct:

```go
switch i {
case Succeeded, Queued:
  return true
}
return false

Reply:
Gotcha, thanks for clarifying.